visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,39 +1,46 @@
-import { FileValidator, CCValidationResult } from "./util/fileValidator"
+import { validate, CCValidationResult } from "./util/fileValidator"
 import { AttributeTypes, CCFile, NameDataPair, BlacklistType, BlacklistItem } from "./codeCharta.model"
 import _ from "lodash"
 import { NodeDecorator } from "./util/nodeDecorator"
+import { OldAttributeTypes, ExportBlacklistType, ExportCCFile } from "./codeCharta.api.model"
 import { StoreService } from "./state/store.service"
 import { addFile, setSingle } from "./state/store/files/files.actions"
 import { getCCFiles } from "./model/files/files.helper"
+import { DialogService } from "./ui/dialog/dialog.service"
 
 export class CodeChartaService {
 	public static ROOT_NAME = "root"
 	public static ROOT_PATH = "/" + CodeChartaService.ROOT_NAME
 	public static readonly CC_FILE_EXTENSION = ".cc.json"
 
-	constructor(private storeService: StoreService) {}
+	constructor(private storeService: StoreService, private dialogService: DialogService) {}
 
 	public loadFiles(nameDataPairs: NameDataPair[]): Promise<void> {
 		return new Promise((resolve, reject) => {
 			nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-				const validationResult: CCValidationResult = FileValidator.validate(nameDataPair.content)
-				if (validationResult.error.length === 0) {
-					const ccFile = this.getCCFile(nameDataPair.fileName, nameDataPair.content)
-					NodeDecorator.preDecorateFile(ccFile)
-					this.storeService.dispatch(addFile(ccFile))
-					if (validationResult.warning.length !== 0) {
-						reject(validationResult)
+				try {
```

Comment:
This is independent from the change: this function is purely sync code or promises get a new chain. Thus, wrapping the code in `new Promise` seems wrong.

Reply:
The promise chain in this one here is a mess. I'm not even sure if we need promises at all. I'll revisit that for sure.